Tests/Integration/PauseTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+namespace Resque\Integration\Plugins\Tests;
+
+use Resque;
+use Resque\Plugins\Pause;
+use PHPUnit_Framework_TestCase;
+
+/**
+ * Pause tests.
+ *
+ * @package     PHP Resque Pause
+ * @author      Wedy Chainy <askwedi@wedipedia.org>
+ * @license     http://www.opensource.org/licenses/mit-license.php
+ */
+class PauseTest extends PHPUnit_Framework_TestCase
+{
+    /** @var Pause */
+    protected $pauser = null;
+
+    public static function setUpBeforeClass()
+    {
+        $testMisc = realpath(__DIR__ . '/misc/');
+        $redisConf = "$testMisc/redis.conf";
+
+        // Attempt to start our own redis instance for tesitng.
+        exec('which redis-server', $output, $returnVar);
+        if ($returnVar != 0) {
+            echo "Cannot find redis-server in path. Please make sure redis is installed.\n";
+            exit(1);
+        }
+
+        exec("cd $testMisc; redis-server $redisConf", $output, $returnVar);
+        usleep(500000);
+        if ($returnVar != 0) {
+            echo "Cannot start redis-server.\n";
+            exit(1);
+        }
+
+        // Get redis port from conf
+        $config = file_get_contents($redisConf);
+        if (!preg_match('#^\s*port\s+([0-9]+)#m', $config, $matches)) {
+            echo "Could not determine redis port from redis.conf";
+            exit(1);
+        }
+
+        Resque::setBackend('localhost:' . $matches[1]);
+    }
+
+    public function setUp()
+    {
+        Resque::redis()->flushAll();
+        $this->pauser = new Pause();
+    }
+
+    public function testPause()
+    {
+        Resque::enqueue('upgrade:test', 'test');
+        Resque::enqueue('upgrade:test', 'test');
+        $this->assertEquals(2, Resque::size('upgrade:test'));
+        $this->assertTrue($this->pauser->pause('upgrade:test'));
+        $this->assertEquals(0, Resque::size('upgrade:test'));
+    }
+
+    public function testPauseNonExistentQueue()
+    {
+        $this->assertFalse($this->pauser->pause('upgrade:test2'));
+    }
+
+    public function testResume()
+    {
+        Resque::enqueue('upgrade:test3', 'test');
+        Resque::enqueue('upgrade:test3', 'test');
+        $this->pauser->pause('upgrade:test3');
+        $this->assertTrue($this->pauser->resume('upgrade:test3'));
+        $this->assertEquals(2, Resque::size('upgrade:test3'));
+    }
+
+    public function testResumeButNeverBeenPausedBefore()
+    {
+        Resque::enqueue('upgrade:test4', 'test');
+        Resque::enqueue('upgrade:test4', 'test');
+        $this->assertFalse($this->pauser->resume('upgrade:test4'));
+    }
+
+    public function testIsPaused()
+    {
+        Resque::enqueue('upgrade:test5', 'test');
+        $this->pauser->pause('upgrade:test5');
+        $this->assertTrue($this->pauser->isPaused('upgrade:test5'));
+    }
+
+    public function testIsPaused2()
```

Comment:
Really long names > `testIsPaused{,2,3}`. Do these tests make sense to each be in isolation?